### PR TITLE
Add C API to query the type of a JS value

### PR DESCRIPTION
**Pull request by galpeter**
New API function:
 * jerry_value_get_type

### CODE REVIEW

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -764,6 +764,35 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
...
+  jerry_assert_api_available ();
+
+  jerry_type_t return_type;
+  lit_magic_string_id_t lit_id = ecma_get_typeof_lit_id (jerry_get_arg_value (value));
+
+  JERRY_ASSERT (lit_id != LIT_MAGIC_STRING__EMPTY);
+
+  switch (lit_id)
+  {
+    case LIT_MAGIC_STRING_UNDEFINED: return_type = JERRY_TYPE_UNDEFINED;  break;
```
Comment (LaszloLango) [inline jerry-core/api/jerry.c]: What about JERRY_TYPE_NULL?

Comment (galpeter) [reply jerry-core/api/jerry.c]: As the ecma 262 5.1 reference describes the NULL in JS as object. However, I'm in favour of having a JERRY_TYPE_NULL, but then the typeof name could be a bit misleading.

Comment (LaszloLango) [reply jerry-core/api/jerry.c]: I see. Leave as it is.

**Comment by galpeter** [comment]
I'll change this to have a bit different name and meaning than the 'typeof' operation to allow the null type to be returned. Stay tuned.

**Comment by galpeter** [comment]
@LaszloLango I've updated the PR

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -766,6 +766,53 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
...
+
+  jerry_type_t return_type;
+  jerry_value_t argument = jerry_get_arg_value (value);
+  lit_magic_string_id_t lit_id = ecma_get_typeof_lit_id (argument);
+
+  JERRY_ASSERT (lit_id != LIT_MAGIC_STRING__EMPTY);
+
+  switch (lit_id)
+  {
+    case LIT_MAGIC_STRING_UNDEFINED:
```
Comment (zherczeg) [inline jerry-core/api/jerry.c]: You have to put `{}` around cases, as the style checker also complains about it.

File: jerry-core/api/jerry.c [hunk 3]
```diff
@@ -766,6 +766,63 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
...
+    {
+      return_type = JERRY_TYPE_UNDEFINED;
+      break;
```
Comment (zherczeg) [inline jerry-core/api/jerry.c]: please use return

**Comment by galpeter** [comment]
@zherczeg I've updated the PR, please check.

File: jerry-core/ecma/base/ecma-helpers.h [hunk 4]
```diff
@@ -187,6 +187,7 @@ void ecma_value_assign_number (ecma_value_t *value_p, ecma_number_t ecma_number)
 void ecma_free_value (ecma_value_t value);
 void ecma_fast_free_value (ecma_value_t value);
 void ecma_free_value_if_not_object (ecma_value_t value);
+lit_magic_string_id_t ecma_get_typeof_lit_id (ecma_value_t value);
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-helpers.h]: One more minor thing: since this is not typeof, lets rename this function to `ecma_get_value_type_lit_id`.

Comment (galpeter) [reply jerry-core/ecma/base/ecma-helpers.h]: This function implements the typeof function as this was extracted from the op_typeof call. I'm using this in both places.

**Pull request merged**